bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/MathUtils.java

```diff
@@ -63,4 +67,28 @@ public static long remainder(long numerator, long denominator) {
             }
         }
     }
+
+    public static long addExact(long num1, long num2) {
+        long result = num1 + num2;
+        if (((num1 ^ result) & (num2 ^ result)) < 0) {
+            throwIntRangeOverflowError();
+        }
+        return result;
+    }
+
+    public static long subtractExact(long num1, long num2) {
+        long result = num1 - num2;
+        if (((num1 ^ num2) & (num1 ^ result)) < 0) {
+            throwIntRangeOverflowError();
+        }
+        return result;
+    }
+
+    public static long multiplyExact(long num1, long num2) {
+        long result = num1 * num2;
+        if ((num1 != 0 && result / num1 != num2) || (num2 != 0 && result / num2 != num1)) {
```

Comment:
why did we decide not to use the `Math.multiplyExact`? Is it to reduce function calls?

Reply:
I checked for performance using hyperfine using 5000 runs of a collection of multiplications, this one gave the best performance out of 3 methods : this one : Math.multiplyExact , and checking whether
if ((num1 != 0 && result / num1 != num2) || (num1==-1L && num2==-9223372036854775808L))